CTOR-737 : Refresh operatingsystems-mac-snmp (#3495)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/operatingsystems-mac-snmp.md

id: operatingsystems-mac-snmp
title: Mac SNMP
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack assets

### Templates

The Monitoring Connector **Mac SNMP** brings a host template:

* **OS-Mac-SNMP-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="OS-Mac-SNMP-custom" label="OS-Mac-SNMP-custom">

| Service Alias | Service Template           | Service Description                                                                                                                                |
|:--------------|:---------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------|

> | Cpu           | OS-Mac-Cpu-SNMP-custom     | Check the rate of utilization of CPU for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPU |

Reviewer: ```suggestion
| Cpu           | OS-Mac-Cpu-SNMP-custom     | Check the rate of utilization of CPUs for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPUs |
```

> | Disk-IO       | OS-Mac-Disk-IO-SNMP-custom | Check access disk of the disk. For each check the name of the disk will appear "label" rather than the letter assigned                             |

Reviewer: ```suggestion
| Disk-IO       | OS-Mac-Disk-IO-SNMP-custom | Check disk access. For each check the name of the disk will appear as a "label" rather than the letter assigned                             |
```

Reviewer: as a label?

| Load          | OS-Mac-Load-SNMP-custom    | Check the server load average                                                                                                                      |

> | Memory        | OS-Mac-Memory-SNMP-custom  | Check the rate of the utilization of memory                                                                                                        |

Reviewer: ```suggestion
| Memory        | OS-Mac-Memory-SNMP-custom  | Check the rate of memory usage |
```

| Swap          | OS-Mac-Swap-SNMP-custom    | Check virtual memory usage                                                                                                                         |

> The services listed above are created automatically when the **OS-Mac-SNMP-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias        | Service Template                        | Service Description                                                                               | Discovery  |
|:---------------------|:----------------------------------------|:--------------------------------------------------------------------------------------------------|:----------:|

> | Disk-Generic-Id      | OS-Mac-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each checks the name of the disk will appear        |            |

Reviewer: ```suggestion
| Disk-Generic-Id      | OS-Mac-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each check the name of the disk will appear        |            |
```

> | Disk-Generic-Name    | OS-Mac-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each checks the mount pont of the disk will appear  |            |

Reviewer: ```suggestion
| Disk-Generic-Name    | OS-Mac-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each check the mount point of the disk will appear  |            |
```

> | Disk-Global          | OS-Mac-Disk-Global-SNMP-custom          | Check the rate of free space on disks. For each checks the mount point of disks will appear       | X          |

Reviewer: ```suggestion
| Disk-Global          | OS-Mac-Disk-Global-SNMP-custom          | Check the rate of free space on disks. For each check the mount point of disks will appear       | X          |
```

> | Disk-Generic-Id      | OS-Mac-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each check the name of the disk will appear        |            |
> | Disk-Generic-Name    | OS-Mac-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each checks the mount pont of the disk will appear  | X          |
> | Disk-Global          | OS-Mac-Disk-Global-SNMP-custom          | Check the rate of free space on disks. For each checks the mount point of disks will appear       |            |
> | Inodes-Global        | OS-Mac-Inodes-Global-SNMP-custom        | Check Inodes space usage on partitions                                                            |            |
> | Process-Generic      | OS-Mac-Process-Generic-SNMP-custom      | Check Unix process/service is working                                                             |            |
> | Traffic-Generic-Id   | OS-Mac-Traffic-Generic-Id-SNMP-custom   | Check the bandwidth of the interface. For each checks the name of the interface will appear       |            |
> | Traffic-Generic-Name | OS-Mac-Traffic-Generic-Name-SNMP-custom | Check the bandwidth of the interface. For each checks the name of the interface will appear       | X          |
> | Traffic-Global       | OS-Mac-Traffic-Global-SNMP-custom       | Check the bandwidth of interfaces. For each checks the name of the interface will appear          |            |
> | Uptime               | OS-Mac-Uptime-SNMP-custom               | Time since the server has been working and available                                              |            |

Author: ```suggestion
| Disk-Generic-Id      | OS-Mac-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each check the name of the disk will appear        |            |
| Disk-Generic-Name    | OS-Mac-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each checks the mount pont of the disk will appear  |            |
| Disk-Global          | OS-Mac-Disk-Global-SNMP-custom          | Check the rate of free space on disks. For each checks the mount point of disks will appear       | X          |
| Inodes-Global        | OS-Mac-Inodes-Global-SNMP-custom        | Check Inodes space usage on partitions                                                            |            |
| Process-Generic      | OS-Mac-Process-Generic-SNMP-custom      | Check Unix process/service is working                                                             |            |
| Traffic-Generic-Id   | OS-Mac-Traffic-Generic-Id-SNMP-custom   | Check the bandwidth of the interface. For each checks the name of the interface will appear       |            |
| Traffic-Generic-Name | OS-Mac-Traffic-Generic-Name-SNMP-custom | Check the bandwidth of the interface. For each checks the name of the interface will appear       |            |
| Traffic-Global       | OS-Mac-Traffic-Global-SNMP-custom       | Check the bandwidth of interfaces. For each checks the name of the interface will appear          |  X         |
| Uptime               | OS-Mac-Uptime-SNMP-custom               | Time since the server has been working and available                                              |            |
```

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_mac_snmp.pl \
	--plugin=os::mac::snmp::plugin \
	--mode=interfaces \
	--hostname=10.0.0.1 \
	--snmp-version='2c' \
	--snmp-community='my-snmp-community'  \
	--interface='.*' \
	--name \
	--add-status \
	--add-traffic \
	--critical-status='' \
	--warning-in-traffic='80' \
	--critical-in-traffic='90' \
	--warning-out-traffic='80' \
	--critical-out-traffic='90' \
	--verbose
```

The expected command output is shown below:

```bash

> OK: All interfaces are ok | '*interface_name*#status'='up';;;;'*interface_name*#interface.traffic.in.bitspersecond'=20b/s;;;;'*interface_name*#interface.traffic.out.bitspersecond'=20b/s;;;;

Author: ```suggestion
OK: All interfaces are ok | '*interface_name*#status'='up';;;;'*interface_name*#interface.traffic.in.bitspersecond'=20b/s;80;90;;'*interface_name*#interface.traffic.out.bitspersecond'=20b/s;80;90;;
```

> OK: All interfaces are ok | '*interface_name*#status'='up';;;;'*interface_name*#interface.traffic.in.bitspersecond'=20b/s;;;;'*interface_name*#interface.traffic.out.bitspersecond'=20b/s;;;;

Reviewer: same as FR
